### PR TITLE
#177: setup of administration frontend

### PR DESCRIPTION
**Pull request by aiquita**
- create react app
- added graphql support
- added blueprintjs
- created navigation skeleton

### CODE REVIEW

**Review by elKei24 — APPROVED** [review]

File: administration/public/index.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (Schedulaar) [inline administration/public/index.html:27]:
```suggestion
    <title>Ehrenamtskarte Verwaltung</title>
```

File: administration/public/index.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
```
Comment (Schedulaar) [inline administration/public/index.html:10]:
```suggestion
      content="Verwaltung der Ehrenamtskarte"
```

File: administration/public/robots.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# https://www.robotstxt.org/robotstxt.html
+User-agent: *
+Disallow:
```
Comment (Schedulaar) [inline administration/public/robots.txt:3]:
```suggestion
Disallow: /
```

File: administration/src/App.test.tsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
+  render(<App />);
+  const linkElement = screen.getByText(/learn react/i);
+  expect(linkElement).toBeInTheDocument();
+});
```
Comment (Schedulaar) [inline administration/src/App.test.tsx:9]: Can be removed; probably fails anyway?

File: administration/src/components/Navigation.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import {Button, Navbar} from "@blueprintjs/core";
+
+export class Navigation extends React.Component<any, any> {
```
Comment (Schedulaar) [inline administration/src/components/Navigation.tsx:4]: We should avoid using `any`

File: administration/src/logo.svg [hunk 6]
```diff
@@ -0,0 +1 @@
...21.2 430c-9.3-9.6-18.6-20.3-27.8-32 9 .4 18.2.7 27.5.7 9.4 0 18.7-.2 27.8-.7-9 11.7-18.3 22.4-27.5 32zm-74.4-58.9c-14.2-2.1-27.9-4.7-41-7.9 3.7-12.9 8.3-26.2 13.5-39.5 4.1 8 8.4 16 13.1 24 4.7 8 9.5 15.8 14.4 23.4zM420.7 163c9.3 9.6 18.6 20.3 27.8 32-9-.4-18.2-.7-27.5-.7-9.4 0-18.7.2-27.8.7 9-11.7 18.3-22.4 27.5-32zm-74 58.9c-4.9 7.7-9.8 15.6-14.4 23.7-4.6 8-8.9 16-13 24-5.4-13.4-10-26.8-13.8-39.8 13.1-3.1 26.9-5.8 41.2-7.9zm-90.5 125.2c-35.4-15.1-58.3-34.9-58.3-50.6 0-15.7 22.9-35.6 58.3-50.6 8.6-3.7 18-7 27.7-10.1 5.7 19.6 13.2 40 22.5 60.9-9.2 20.8-16.6 41.1-22.2 60.6-9.9-3.1-19.3-6.5-28-10.2zM310 490c-13.6-7.8-19.5-37.5-14.9-75.7 1.1-9.4 2.9-19.3 5.1-29.4 19.6 4.8 41 8.5 63.5 10.9 13.5 18.5 27.5 35.3 41.6 50-32.6 30.3-63.2 46.9-84 46.9-4.5-.1-8.3-1-11.3-2.7zm237.2-76.2c4.7 38.2-1.1 67.9-14.6 75.8-3 1.8-6.9 2.6-11.5 2.6-20.7 0-51.4-16.5-84-46.6 14-14.7 28-31.4 41.3-49.9 22.6-2.4 44-6.1 63.6-11 2.3 10.1 4.1 19.8 5.2 29.1zm38.5-66.7c-8.6 3.7-18 7-27.7 10.1-5.7-19.6-13.2-40-22.5-60.9 9.2-20.8 16.6-41.1 22.2-60.6 9.9 3.1 19.3 6.5 28.1 10.2 35.4 15.1 58.3 34.9 58.3 50.6-.1 15.7-23 35.6-58.4 50.6zM320.8 78.4z"/><circle cx="420.9" cy="296.5" r="45.7"/><path d="M520.5 78.1z"/></g></svg>
```
Comment (Schedulaar) [inline administration/src/logo.svg:1]: You could replace this icon with our actual icon;
https://github.com/ehrenamtskarte/ehrenamtskarte/blob/main/frontend/assets/icon/icon.png

**Review by Schedulaar — CHANGES_REQUESTED** [review]
I think you accidentally added a package.json to the root folder (and another one to the `administration` folder). You can remove the one (and the package-lock.json) of the root folder

File: administration/src/components/Navigation.tsx (hunk 5 above)
Comment (aiquita) [reply administration/src/components/Navigation.tsx:4]: Well, you are right, but this code does not have any purpose other than displaying a skeleton.

File: administration/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
+# Getting Started with Create React App
```
Comment (maxammann) [inline administration/README.md:1]: I think this readme can be removed.

**Review by maxammann — COMMENTED** [review]
Should we name this really administration? I mean the users of this backend are not "administrators" but cities. We (the administrators) are not using this. Maybe a more generic term would fit like "web-frontend"

File: administration/src/components/Navigation.tsx (hunk 5 above)
Comment (maxammann) [inline administration/src/components/Navigation.tsx:4]:
```suggestion
export class Navigation extends React.Component<{}, {}> {
```

**Review by maxammann — APPROVED** [review]

**Comment by Schedulaar** [comment]
@maxammann Imo web-frontend rather sounds like a web-frontend for the app (facing "otto-normal" users). Administration is fine for me; maybe also content-management or just management would suit as well...

**Review by Schedulaar — APPROVED** [review]

**Pull request merged**